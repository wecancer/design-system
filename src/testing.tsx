import React from 'react'
import {render, RenderResult} from '@testing-library/react'
import WecancerProvider from './WecancerProvider'

type Props = {
  children: React.ReactNode
}

const AllTheProviders = ({children}: Props) => {
  return <WecancerProvider>{children}</WecancerProvider>
}

// eslint-disable-next-line
const appRender = (ui: React.ReactElement, options?: any): RenderResult =>
  render(ui, {wrapper: AllTheProviders, ...options})

// re-export everything
export * from '@testing-library/react'

// override render method
export {appRender as render}
