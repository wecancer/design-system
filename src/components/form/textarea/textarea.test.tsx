import { render, fireEvent } from '../../../testing'

import Textarea from '.'

describe('<Textarea />', () => {
  it('should render snapshot correctly', () => {
    const { container } = render(
      <>
        <Textarea
          value="Text message"
          label="Text label"
          onChange={() => null}
        />
        <Textarea
          value="Text message"
          label="Text label"
          onChange={() => null}
          rows={4}
        />
        <Textarea
          infoType="error"
          label="error label"
          value="error message"
          infoMessage="error message"
          onChange={() => null}
        />
        <Textarea
          infoType="success"
          label="success label"
          value="success message"
          infoMessage="success message"
          onChange={() => null}
        />
        <Textarea
          infoType="caption"
          label="caption label"
          value="caption message"
          infoMessage="caption message"
          onChange={() => null}
        />
      </>,
    )
    expect(container.firstChild).toMatchSnapshot()
  })

  it('should have a button to clear the textarea if the user want', () => {
    const handleClick = jest.fn()
    const { container } = render(
      <Textarea
        value="Testing message..."
        onChange={() => null}
        onClear={handleClick}
      />,
    )
    const buttonClear = container.querySelector('[data-icon-name="times"]')
      ?.parentNode as HTMLButtonElement

    fireEvent.click(buttonClear)

    expect(handleClick).toHaveBeenCalledTimes(1)
  })

  it('should have an onChange event on component textarea', () => {
    const { container } = render(
      <Textarea
        value=""
        onChange={({ value, event }) => {
          expect(value).toBe('testing change...')
          expect(event?.currentTarget.value).toBe('testing change...')
        }}
      />,
    )
    const textarea = container.querySelector('textarea') as HTMLElement
    fireEvent.change(textarea, { target: { value: 'testing change...' } })
  })
})
