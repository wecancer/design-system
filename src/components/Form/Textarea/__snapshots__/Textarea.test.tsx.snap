// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Textarea /> should render snapshot correctly 1`] = `
.c5 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c5 svg {
  width: 1.375rem;
  height: 1.375rem;
}

.c3 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  padding: 0;
  outline: none;
  border: none;
  background: transparent;
}

.c0 {
  position: relative;
  font-family: Poppins,sans-serif;
}

.c1 {
  width: 100%;
  font-size: 1rem;
  padding: 1rem 1.5rem;
  box-sizing: border-box;
  font-family: Poppins,sans-serif;
  border-radius: 1rem;
  outline: none;
  background-color: #FCFCFC;
  border: 1px solid #A0A3BD;
}

.c1:focus {
  border-color: #14142B;
}

.c1:focus + label {
  color: #6E7191;
  font-size: 0.875rem;
  top: -0.6875rem;
}

.c1 + label {
  color: #6E7191;
  font-size: 0.875rem;
  top: -0.6875rem;
}

.c4 {
  position: absolute;
  top: 1rem;
  right: 1rem;
  cursor: pointer;
}

.c2 {
  position: absolute;
  top: 1rem;
  left: 1.25rem;
  padding: 0 0.25rem;
  background: transition:all 250ms ease;
  background: linear-gradient( 180deg,rgba(255,255,255,0) 0%,rgba(20,20,255,0) 50%,#FCFCFC 50%,#FCFCFC 100% );
}

<div
  class="c0"
>
  <textarea
    class="c1"
    rows="2"
  >
    Text message
  </textarea>
  <label
    class="c2"
  >
    Text label
  </label>
  <button
    class="c3 c4"
    type="button"
  >
    <div
      class="c5"
      data-icon-name="times"
    >
      <span />
    </div>
  </button>
</div>
`;
